arbnode/transaction_streamer.go

```diff
@@ -70,32 +74,58 @@ type TransactionStreamer struct {
 	broadcastServer *broadcaster.Broadcaster
 	inboxReader     *InboxReader
 	delayedBridge   *DelayedBridge
+	espressoClient  *espressoClient.Client
+
+	pendingTxnsQueueMutex sync.Mutex // cannot be acquired while reorgMutex is held
+	pendingTxnsPos        []arbutil.MessageIndex
+	submittedTxnPos       *arbutil.MessageIndex
+	submittedTxHash       *espressoTypes.TaggedBase64
 }
 
 type TransactionStreamerConfig struct {
 	MaxBroadcasterQueueSize int           `koanf:"max-broadcaster-queue-size"`
 	MaxReorgResequenceDepth int64         `koanf:"max-reorg-resequence-depth" reload:"hot"`
 	ExecuteMessageLoopDelay time.Duration `koanf:"execute-message-loop-delay" reload:"hot"`
+
+	// Espresso specific fields
+	SovereignSequencerEnabled   bool          `koanf:"sovereign-sequencer-enabled"`
+	HotShotUrl                  string        `koanf:"hotshot-url"`
+	EspressoNamespace           uint64        `koanf:"espresso-namespace"`
+	EspressoTimeout             time.Duration `koanf:"espresso-timeout"`
+	EspressoTxnsPollingInterval time.Duration `koanf:"espresso-txns-polling-interval"`
 }
 
 type TransactionStreamerConfigFetcher func() *TransactionStreamerConfig
 
 var DefaultTransactionStreamerConfig = TransactionStreamerConfig{
-	MaxBroadcasterQueueSize: 50_000,
-	MaxReorgResequenceDepth: 1024,
-	ExecuteMessageLoopDelay: time.Millisecond * 100,
+	MaxBroadcasterQueueSize:     50_000,
+	MaxReorgResequenceDepth:     1024,
+	ExecuteMessageLoopDelay:     time.Millisecond * 100,
+	SovereignSequencerEnabled:   false,
+	HotShotUrl:                  "",
+	EspressoTimeout:             time.Second * 5,
+	EspressoTxnsPollingInterval: time.Millisecond * 100,
 }
 
 var TestTransactionStreamerConfig = TransactionStreamerConfig{
-	MaxBroadcasterQueueSize: 10_000,
-	MaxReorgResequenceDepth: 128 * 1024,
-	ExecuteMessageLoopDelay: time.Millisecond,
+	MaxBroadcasterQueueSize:     10_000,
+	MaxReorgResequenceDepth:     128 * 1024,
+	ExecuteMessageLoopDelay:     time.Millisecond,
+	SovereignSequencerEnabled:   false,
+	HotShotUrl:                  "",
+	EspressoTimeout:             time.Second * 5,
```

Comment:
I can't find any usage of this, but maybe my editor is wrong?

Reply:
Its not used, will remove this